tests/laplace.hpp

```diff
@@ -38,9 +43,10 @@ class Laplace
 public:
   Laplace(MPI_Comm const &comm, unsigned int fe_degree);
 
-  void setup_system(int num_refinements = 3);
+  void setup_system(boost::property_tree::ptree const ptree);
```

Comment:
Would it not be better to add parameters to the constructor arguments, store them and then use for `setup_system` and `assemble_system`?

Reply:
I have no opinion on this. `ptree` is only used in `setup_system` and `MaterialProperty` in `assemble_system`. I can add them in the constructor if you prefer.